File: api/src/Entity/InjectStatus.php
<?php

namespace App\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity()
 * @ORM\Table(name="injects_statuses")
 */
class InjectStatus
{
    /**
     * @ORM\Id
     * @ORM\Column(type="string")
     * @ORM\GeneratedValue(strategy="UUID")
     */
    protected $status_id;

    /**
     * @ORM\OneToOne(targetEntity="Inject", inversedBy="inject_status")
     * @ORM\JoinColumn(name="status_inject", referencedColumnName="inject_id", onDelete="CASCADE")
     */
    protected $status_inject;

    /**
     * @ORM\Column(type="string", nullable=true)
     */
    protected $status_name;

    /**
     * @ORM\Column(type="text", nullable=true)
     */
    protected $status_message;

    /**
     * @ORM\Column(type="datetime", nullable=true)
     */
    protected $status_date;

    /**
     * @ORM\Column(type="integer", nullable=true)
     */
    protected $status_execution;

    public function getStatusId()
    {
        return $this->status_id;
    }

    public function setStatusId($id)
    {
        $this->status_id = $id;
        return $this;
    }

    public function getStatusInject()
    {
        return $this->status_inject;
    }

    public function setStatusInject($inject)
    {
        $this->status_inject = $inject;
        return $this;
    }

    public function getStatusName()
    {
        return $this->status_name;
    }

    public function setStatusName($name)
    {
        $this->status_name = $name;
        return $this;
    }

    public function getStatusMessage()
    {
        return $this->status_message;
    }

    public function setStatusMessage($message)
    {
        $this->status_message = $message;
        return $this;
    }

    public function getStatusDate()
    {
        return $this->status_date;
    }

    public function setStatusDate($date)
    {
        $this->status_date = $date;
        return $this;
    }

    public function getStatusExecution()
    {
        return $this->status_execution;
    }

    public function setStatusExecution($status_execution)
    {
        $this->status_execution = $status_execution;
        return $this;
    }
}

File: openex-api/src/main/java/io/openex/database/model/InjectStatus.java
package io.openex.database.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.openex.database.converter.StatusReportingConverter;
import io.openex.model.Execution;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.time.Instant;
import java.util.Objects;

import static java.time.Instant.now;

@Entity
@Table(name = "injects_statuses")
public class InjectStatus implements Base {
    @Id
    @Column(name = "status_id")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @JsonProperty("status_id")
    private String id;

    @Column(name = "status_name")
    @JsonProperty("status_name")
    private String name;

    @Column(name = "status_message")
    @Convert(converter = StatusReportingConverter.class)
    @JsonProperty("status_message")
    private StatusReporting reporting;

    @Column(name = "status_date")
    @JsonProperty("status_date")
    private Instant date;

    @Column(name = "status_execution")
    @JsonProperty("status_execution")
    private Integer executionTime;

    @OneToOne
    @JoinColumn(name = "status_inject")
    @JsonIgnore
    private Inject<?> inject;

    // region transient
    public static <T> InjectStatus fromExecution(Execution execution, Inject<T> inject) {
        InjectStatus injectStatus = new InjectStatus();
        injectStatus.setInject(inject);
        injectStatus.setDate(now());
        injectStatus.setExecutionTime(execution.getExecution());
        injectStatus.setName(execution.getStatus().name());
        injectStatus.setReporting(execution.getReporting());
        return injectStatus;
    }
    // endregion

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public StatusReporting getReporting() {
        return reporting;
    }

    public void setReporting(StatusReporting reporting) {
        this.reporting = reporting;
    }

    public Instant getDate() {
        return date;
    }

    public void setDate(Instant date) {
        this.date = date;
    }

    public Integer getExecutionTime() {
        return executionTime;
    }

    public void setExecutionTime(Integer executionTime) {
        this.executionTime = executionTime;
    }

    public Inject<?> getInject() {
        return inject;
    }

    public void setInject(Inject<?> inject) {
        this.inject = inject;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || !Base.class.isAssignableFrom(o.getClass())) return false;
        Base base = (Base) o;
        return id.equals(base.getId());
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}
